Add Cookstyle 6.1 release notes (#593)
chef/cookstyle: RELEASE_NOTES.md

## Cookstyle 6.1

### 5 New Cops

#### ChefModernize/NodeRolesInclude

The `ChefModernize/NodeRolesInclude` cop detects cookbooks that use `node['roles'].include?('foo')` to see if a particular role has been applied to a node. The simpler `node.role?('foo')` helper should be used instead.

`Enabled by default`: True

`Autocorrects`: Yes

#### ChefDeprecations/PowershellCookbookHelpers

> The `ChefDeprecations/PowershellCookbookHelpers` cop detects cookbooks that use the deprecated `Powershell::VersionHelper.powershell_version?` helper from the `powershell` cookbook. Chef Infra Client provides information on the installed PowerShell release at `node['powershell']['version']` and Chef Infra Client 15.8 and later includes a new `powershell_version` which should be used when possible.

Reviewer: ```suggestion
The `ChefDeprecations/PowershellCookbookHelpers` cop detects cookbooks that use the deprecated `Powershell::VersionHelper.powershell_version?` helper from the `powershell` cookbook. Chef Infra Client provides information on the installed PowerShell release at `node['powershell']['version']` and Chef Infra Client 15.8 and later includes a new `powershell_version` method which should be used when possible.
```

`Enabled by default`: True

`Autocorrects`: Yes

#### ChefCorrectness/ConditionalRubyShellout

> The `ChefCorrectness/ConditionalRubyShellout` detects resources that use Ruby code to shellout in not_if and only_if conditionals when they can just shellout without the additional complexity of Ruby.

Reviewer: ```suggestion
The `ChefCorrectness/ConditionalRubyShellout` cop detects resources that use unnecessary Ruby code in `not_if` and `only_if` conditionals to shellout when a string can be used to avoid additional complexity.
```

This sentence read weirdly to me and this was my best attempt to make it sound less awkward.

Shelling out with ruby:
```ruby
cookbook_file '/logs/foo/error.log' do
  only_if { system('wget https://www.bar.com/foobar.txt -O /dev/null') }
end
```

Shelling out without Ruby:
```ruby
cookbook_file '/logs/foo/error.log' do
  only_if 'wget https://www.bar.com/foobar.txt -O /dev/null'
end
```

`Enabled by default`: True

`Autocorrects`: Yes

#### ChefSharing/IncludePropertyDescriptions

> The `ChefSharing/IncludePropertyDescriptions` cop detects resource properties that don't include a description value. These descriptions are used by the `chef-resource-inspector` command and can be used to automatically generate documentation.

Reviewer 2: ```suggestion
The `ChefSharing/IncludePropertyDescriptions` cop detects resource properties that don't include a description value. The `chef-resource-inspector` command uses the description in resources properties and that description can be used to automatically generate documentation.
```

> The `ChefSharing/IncludeResourceDescriptions` cop detects resource properties that don't include a description value. The description in resources properties are used by the `chef-resource-inspector` command and can be used to automatically generate documentation.

Reviewer: ```suggestion
The `ChefSharing/IncludePropertyDescriptions` cop detects resource properties that don't include a description value. These descriptions are used by the `chef-resource-inspector` command and can be used to automatically generate documentation.
```

`Enabled by default`: False

`Autocorrects`: No

#### ChefSharing/IncludeResourceDescriptions

> The `ChefSharing/IncludeResourceDescriptions` cop detects resource that don't include a description. The description in resources is used by the `chef-resource-inspector` command and can be used to automatically generate documentation.

Reviewer: ```suggestion
The `ChefSharing/IncludeResourceDescriptions` cop detects resources that don't include a description. The description in resources is used by the `chef-resource-inspector` command and can be used to automatically generate documentation.
```

> The `ChefSharing/IncludeResourceDescriptions` cop detects resource that don't include a description. The description in resources is used by the `chef-resource-inspector` command and can be used to automatically generate documentation.

Reviewer 2: ```suggestion
The `ChefSharing/IncludeResourceDescriptions` cop detects resource that don't include a description. The `chef-resource-inspector` command uses the description in resources properties and that description can be used to automatically generate documentation.
```

Author: Pulled in manually
